packager/media/formats/mp2t/ts_writer.h

```diff
@@ -67,6 +72,7 @@ class TsWriter {
   std::unique_ptr<ProgramMapTableWriter> pmt_writer_;
 
   std::unique_ptr<File, FileCloser> current_file_;
+  std::unique_ptr<BufferWriter> current_buffer_;
```

Comment:
It does not need to be a unique pointer; also, we can call it |segment_buffer_| i.e. It can be 

```c++
BufferWriter segment_buffer_;
```

I think it will be cleaner to manage the buffer in `TsSegmenter` class. We will need to update the signature of the two functions `NewSegment` and `AddPesPacket` to something like though:

```c++
virtual bool NewSegment(BufferWriter* buffer);
virtual bool AddPesPacket(std::unique_ptr<PesPacket> pes_packet, BufferWriter* buffer);
```

Then we don't need to have the the function `GetPosition` and `WriteFile` in this class.

What do you think?

Reply:
Makes sense. I will make this change.